functests/4_latency/latency.go

```diff
@@ -135,38 +139,28 @@ var _ = Describe("[performance] Latency Test", func() {
 			oslatArgs := []string{
 				fmt.Sprintf("-runtime=%s", latencyTestRuntime),
 			}
-			latencyTestPod = getLatencyTestPod(profile, workerRTNode, testName, oslatArgs)
-			createLatencyTestPod(latencyTestPod)
+			latencyTestPod = getLatencyTestPod(profile, workerRTNode, testName, oslatArgs, logName)
+			createLatencyTestPod(latencyTestPod, workerRTNode, logName)
 
 			// verify the maximum latency only when it requested, because this value can be very different
 			// on different systems
 			if maximumLatency == -1 {
 				Skip("no maximum latency value provided, skip buckets latency check")
 			}
 
-			latencies := extractLatencyValues("oslat", `Maximum:\t*([\s\d]*)\(us\)`, workerRTNode)
-
-			// under the output of the oslat very often we have one anomaly high value, for example
-			// Maximum:    16543 15 15 14 13 12 12 13 12 12 12 12 12 12 12 12 12 (us)
-			// it still unclear if it oslat bug or the kernel one, but we definitely do not want to
-			// fail our test on it
-			var anomaly bool
-			for _, lat := range strings.Split(latencies, " ") {
+			latencies := extractLatencyValues(logName, `Maximum:\t*([\s\d]*)\(us\)`, workerRTNode)
+			latenciesList := strings.Split(latencies, " ")
+			for _, lat := range latenciesList {
 				if lat == "" {
 					continue
 				}
-
 				curr, err := strconv.Atoi(lat)
 				Expect(err).ToNot(HaveOccurred())
 
-				// skip the anomaly value
-				if curr > maximumLatency && !anomaly {
-					anomaly = true
-					continue
-				}
+				Expect(curr < maximumLatency).To(BeTrue(), "The current latency %d is bigger than the expected one %d : \n %s", curr, maximumLatency, getLogFile(workerRTNode, logName))
```

Comment:
`getLogFile` should not be called inline like this, in the error message explaining a failed assertion, because it can fail, like we observed recently. Should always be called explicitely, and we need to handle any errors from it explicitely.
Can be fixed in a later PR.

Reply:
I see , ok will note that & fix it later.Thanks !

Reply:
since I had to make additional change , I have modified this part of code as well .